Source/missiles.cpp

```diff
@@ -1106,19 +1106,19 @@ void AddReflection(Missile &missile, const AddMissileParameter & /*parameter*/)
 {
 	missile._miDelFlag = true;
 
-	if (missile._misource < 0)
+	if (missile.sourceType() != MissileSource::Player)
 		return;
 
-	Player &player = Players[missile._misource];
+	Player *player = missile.sourcePlayer();

```

Comment:
```suggestion
	Player &player = *missile.sourcePlayer();
```
You can dereference the pointer immediately since the type was already checked earlier.